1_fetch.R

```diff
@@ -57,7 +57,36 @@ p1_targets_list <- list(
   tar_target(
     p1_inst_data,
     get_inst_nwis_data(p1_nwis_sites_inst,parameter,start_date=earliest_date,end_date=dummy_date),
-    pattern = map(p1_nwis_sites_inst))
+    pattern = map(p1_nwis_sites_inst)),
   
+  tar_target(
+    p1_reaches_shp_zip,
+    # [Jeff] I downloaded this manually from science base: 
+    # https://www.sciencebase.gov/catalog/item/5f6a285d82ce38aaa244912e
+    # Because it's a shapefile, it's not easily downloaded using sbtools
+    # like other files are (see https://github.com/USGS-R/sbtools/issues/277).
+    # Because of that and since it's small (<700 Kb) I figured it'd be fine to
+    # just include in the repo and have it loosely referenced to the sb item ^
+    "1_fetch/in/study_stream_reaches.zip",
+    format = "file"
+  ),
+  
+  # Unzip zipped shapefile
+  tar_target(
+    p1_reaches_shp_unzipped,
+    {shapedir = "1_fetch/out/study_stream_reaches"
+    unzip(p1_reaches_shp_zip, exdir = shapedir)},
+    format = "file"
+  ),
+  
+  # read shapefile into sf object
+  tar_target(
+    p1_reaches_sf,
+    # third item is the .shp file (all are 'dbf', 'prj', 'shp', and 'shx')
+    {
+    shp_file_idx <- grep(".shp", p1_reaches_shp_unzipped)
```

Comment:
Adding the argument value=TRUE to the `grep()` call would return the name of the shp file that you could pass in to `sf::st_read()` without specifying the index

Reply:
oooh. Thanks! Much cleaner.